### PR TITLE
feat: create separate buckets

### LINKED ISSUE #59: Add separate public and secure upload buckets
It is anticipated that there will be a need for both publicly accessible and fully secured S3 buckets for storing user uploads with different security requirements.  Only one of those currently exists, so the other must be created, with clear delineation between them.

### PR DESCRIPTION
**Pull request by chriscasto**
This PR replaces the single user upload bucket with two separate buckets with different visibility: `public_uploads` and `secure_uploads`.  The `public_uploads` bucket has a policy that permits unauthenticated GETs.  It should be noted that this level of public access is probably only going to be temporary, and that the long-term approach will most likely involve having CloudFront serve requests for content in this bucket.

This PR also creates a bucket for holding static content belonging to the public and partners sites (CSS, JS, images, etc).  This will be used by the app pipeline for storing static and pre-generated content.

Resolves #59

### CODE REVIEW

File: bloom-instance/s3.tf [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+  bucket_prefix = "${var.name_prefix}-secure-uploads"
+}
+
+resource "aws_s3_bucket" "static_content" {
+  bucket_prefix = "${var.name_prefix}-static-content"
+}
+
+# Allow anyone to read from the public uploads bucket
+resource "aws_s3_bucket_policy" "public_uploads" {
+  bucket = aws_s3_bucket.bucket.id
```
Comment (ifranch) [inline bloom-instance/s3.tf:15]: Should this be aws_s3_bucket.public_uploads.id instead since this policy is for the public uploads bucket?

Comment (chriscasto) [reply bloom-instance/s3.tf:15]: Yep, you're right.  Good catch

File: bloom-instance/s3.tf [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Sid = "AllowReadsFromUploadBucket"
+        Action = [
+          "s3:GetObject",
+        ]
+        Effect    = "Allow"
+        Principal = "*"
+        Resource  = "arn:aws:s3:${var.aws_region}::${aws_s3_bucket.public_uploads.bucket}/*",
```
[the remark this answers is not in the document]
Comment (chriscasto) [reply bloom-instance/s3.tf:26]: You're right that `aws_s3_bucket.public_uploads.arn` could have been used here, but in this case the ARN is generated using the bucket name (`aws_s3_bucket.public_uploads.bucket`) with a wildcard at the end to signify that `s3:GetObject` is being granted on all objects within the bucket.

**Comment by chriscasto** [comment]
Fixed some issues and adding the plan output I left off before.

[plan.txt](https://github.com/metrotranscom/doorway-infra/files/11325756/plan.txt)

**Review by ifranch — APPROVED** [review]

**Comment by ifranch** [comment]
> Fixed some issues and adding the plan output I left off before.
> 
> [plan.txt](https://github.com/metrotranscom/doorway-infra/files/11325756/plan.txt)

This doesn't show the private bucket being created. Is there an issue with the command?

**Comment by chriscasto** [comment]
I scoped the plan to the public bucket since that has the policy.  I don't think I can scope it to two separate resources, so I just ran it again without a target.

[plan.txt](https://github.com/metrotranscom/doorway-infra/files/11326099/plan.txt)

**Review by isaac-looker — APPROVED** [review]

**Pull request merged**